test/setup-ca-kubernetes.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+
+# This script generates the certificates needed for securing pmem csi components.
+# This is supposed to run prior to deploying pmem-csi driver in a cluster. 
+
+generate_csr()
+{
+    CN=$1
+    CSR_NAME=${2:-$CN}
+
+    echo "Generating signing request certificate for '$CN'"
+    cert_json=$(echo '{ "CN":  "'$CN'", "key": { "algo": "ecdsa", "size": 256 } }')
+
+    echo $cert_json | cfssl genkey - | cfssljson -bare $CSR_NAME
+
+    echo "Creating kubernetes signing request: $CSR_NAME-csr"
+    kubectl delete csr $CSR_NAME-csr 2> /dev/null
+    cat <<EOF | kubectl create -f -
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: $CSR_NAME-csr
+spec:
+  groups:
+  - system:authenticated
+  request: $(cat $CSR_NAME.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+  - client auth
+EOF
+
+    echo "Approving signing request..."
+    kubectl certificate approve $CSR_NAME-csr
+
+    #retrieve certificate
+    echo "Created : $CSR_NAME.crt"
+    kubectl get csr $CSR_NAME-csr -o jsonpath='{.status.certificate}' | base64 --decode > $CSR_NAME.crt
+}
+
+
+# Directory to use for storing intermediate files
+WORKDIR=${WORKDIR:-/tmp/certs}
+mkdir -p $WORKDIR 2> /dev/null
+PATH="$WORKDIR:$PATH"
+cd $WORKDIR
+
+#download cfssl utilities
+which cfssl cfssljson 2> /dev/null
+if [ $? != 0 ]; then
+    echo "Downloading cfssl tools..."
+    curl -L https://pkg.cfssl.org/R1.2/cfssl_linux-amd64 -o cfssl
+    curl -L https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64 -o cfssljson
+    chmod a+x cfssl cfssljson
+fi
+
+# Generate pmem greistry server certificate signing request
```

Comment:
s/greistry/registry/

Should it be upper case PMEM?

Reply:
Done